### ISSUE TITLE
[BUG] nvstrings test failed : Value mismacth after string to double conversion in ppc64le. 

### ISSUE THREAD

**Issue by pradghos** [rapidsai/cudf#3847]
**Describe the bug**
nvstrings test failed : Value mismacth after string to double conversion in ppc64le. 

Log - 

```
tests]$ pytest -v test_convert.py::test_stod
==================================================================== test session starts =====================================================================
platform linux -- Python 3.6.10, pytest-5.3.2, py-1.8.1, pluggy-0.13.1 -- /opt/anaconda3/envs/gdf/bin/python
cachedir: .pytest_cache
rootdir: /home/builder/cudf/python/nvstrings, inifile: pytest.ini
collected 1 item

test_convert.py::test_stod FAILED                                                                                                                      [100%]

----- 
>       assert_eq(got, expected)

test_convert.py:411:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

arr1 = [1234.0, 5678.0, 90.0, None, -876.0, 543.2, ...], arr2 = [1234.0, 5678.0, 90.0, None, -876.0, 543.2, ...]

    def assert_eq(arr1, arr2):
        if isinstance(arr1, nvstrings.nvstrings):
            arr1 = arr1.to_host()

        if isinstance(arr2, nvstrings.nvstrings):
            arr2 = arr2.to_host()

>       assert np.array_equiv(arr1, arr2)
E       AssertionError

utils.py:22: AssertionError

```

**Steps/Code to reproduce bug**
Below are the steps to reproduce the bug - 

```
>>> s = nvstrings.to_device(
... [
... "-0.12",
... "-.002",
... "2.553",
... "-1.78e+5",
... "-122.33644782"
... ]
... )
>>> got = s.stod()
>>> got
[-0.12000000000000005, -0.0020000000000000005, 2.553000000000001, -177999.99999999994, -122.3364478200001]
```

However, execpected results are like below - 

```
>>> expected = [
... -0.12,
... 2.553,
... -0.002,
... -178000.0,
... -122.33644781999999
... ]
>>> expected
[-0.12, 2.553, -0.002, -178000.0, -122.33644781999999]
>>>
```

**Expected behavior**
After `string` to `double` conversion, it should match with expected results - 

**Environment overview (please complete the following information)**
 - Environment location: It is easy to reproduce in ppc64le environment - 

```
conda list | grep cudf
cudf                      0.11.0          cuda10.2_py36_666.g45906b8
conda list | grep nvstrings
nvstrings                 0.11.0          cuda10.2_py36_626.g9163edb

```

 - Method of cuDF install: cuDF , nvstring 0.11.0 has been built and install in ppc64le platform. 
   

**Environment details**

cudatoolkit version 10.2
gcc 7.3
cudf 0.11.0
nvstrings 0.11.0

**Additional context**
Similar issue is being observed in https://github.com/rapidsai/cudf/issues/3723/

**Comment by rnyak** [comment]
@pradghos I could reproduce the expected results with Rapids 0.11 Stable 
 using `rapidsai/rapidsai:cuda10.1-runtime-ubuntu18.04` container. 

I ran your sample code in the Jupyter Lab, this is what I get:

import nvstrings
s = nvstrings.to_device(["-0.12", "-.002", "2.553", "-1.78e+5", "-122.33644782"])
got = s.stod()
got
[-0.12, -0.002, 2.553, -178000.0, -122.33644781999999]

print(s.stod())
[-0.12, -0.002, 2.553, -178000.0, -122.33644781999999]

Let us know if you can make it work.

**Comment by pradghos** [comment]
> 
> 
> @pradghos I could reproduce the expected results with Rapids 0.11 Stable
> using `rapidsai/rapidsai:cuda10.1-runtime-ubuntu18.04` container.
> 
> I ran your sample code in the Jupyter Lab, this is what I get:
> 
> ```
> import nvstrings
> s = nvstrings.to_device(["-0.12", "-.002", "2.553", "-1.78e+5", "-122.33644782"])
> got = s.stod()
> got
> [-0.12, -0.002, 2.553, -178000.0, -122.33644781999999]
> 
> print(s.stod())
> [-0.12, -0.002, 2.553, -178000.0, -122.33644781999999]
> ```
> 
> Let us know if you can make it work.

@rnyak : Thank for looking - results are coming different than the expected value in the environment - I am trying out - `cuda 10.2`, `ppc64le`, `gcc 7.3` . `RAPIDS 0.11.0`

Another observation is - 
```
>>> s = nvstrings.to_device(["-0.12"])
>>> got = s.stod()
>>> got
[-0.12000000000000005]
>>> s = nvstrings.to_device(["-0.1"])
>>> got = s.stod()
>>> got
[-0.1]
>>>
>>> s = nvstrings.to_device(["0.99"])
>>> got = s.stod()
>>> got
[0.9900000000000003]
>>>
```
whenever there is more than one digit after decimal point , **mantissa** part is getting expanded.

**Comment by galipremsagar** [comment]
This should be resolved by #4141.
